From e6ee42e9bca274b59c42f134f4c71d14ab6ed8a6 Mon Sep 17 00:00:00 2001
From: Nezih Yigitbasi <nyigitbasi@netflix.com>
Date: Tue, 30 Jun 2015 11:00:37 -0700
Subject: [PATCH] PARQUET-316: Fix the benchmark module

`run.sh` is now broken with the packages renamed to `org.apache...` and also somehow the `hadoop-2` profile creates a jar file that doesn't include `/META-INF/BenchmarkList` -- a file that jmh needs:

```
Exception in thread "main" java.lang.RuntimeException: ERROR: Unable to find the resource: /META-INF/BenchmarkList
    at org.openjdk.jmh.runner.AbstractResourceReader.getReaders(AbstractResourceReader.java:96)
    at org.openjdk.jmh.runner.BenchmarkList.find(BenchmarkList.java:104)
    at org.openjdk.jmh.runner.Runner.internalRun(Runner.java:228)
    at org.openjdk.jmh.runner.Runner.run(Runner.java:178)
    at org.openjdk.jmh.Main.main(Main.java:66)
```

Author: Nezih Yigitbasi <nyigitbasi@netflix.com>

Closes #226 from nezihyigitbasi/316 and squashes the following commits:

f9192d5 [Nezih Yigitbasi] PARQUET-316: Fix the benchmark module build instructions and its run script
---
 parquet-benchmarks/README.md | 4 ++--
 parquet-benchmarks/run.sh    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/parquet-benchmarks/README.md b/parquet-benchmarks/README.md
index 646e6baa3e..8da067b09b 100644
--- a/parquet-benchmarks/README.md
+++ b/parquet-benchmarks/README.md
@@ -22,13 +22,13 @@
 First, build the ``parquet-benchmarks`` module
 
 ```
-mvn --projects parquet-benchmarks -amd -DskipTests -Denforcer.skip=true -P hadoop-2 clean package
+mvn --projects parquet-benchmarks -amd -DskipTests -Denforcer.skip=true clean package
 ```
 
 Then, you can run all the benchmarks with the following command
 
 ```
- ./parquet-benchmarks/run.sh -wi 5 -i 5 -f 3 -bm all
+./parquet-benchmarks/run.sh -wi 5 -i 5 -f 3 -bm all
 ```
 
 To understand what each command line argument means and for more arguments please see
diff --git a/parquet-benchmarks/run.sh b/parquet-benchmarks/run.sh
index e92b57db0f..cfaddae1e5 100755
--- a/parquet-benchmarks/run.sh
+++ b/parquet-benchmarks/run.sh
@@ -24,8 +24,8 @@ SCRIPT_PATH=$( cd "$(dirname "$0")" ; pwd -P )
 echo "Starting WRITE benchmarks"
 java -jar ${SCRIPT_PATH}/target/parquet-benchmarks.jar p*Write* "$@"
 echo "Generating test data"
-java -cp ${SCRIPT_PATH}/target/parquet-benchmarks.jar parquet.benchmarks.DataGenerator generate
+java -cp ${SCRIPT_PATH}/target/parquet-benchmarks.jar org.apache.parquet.benchmarks.DataGenerator generate
 echo "Data generated, starting READ benchmarks"
 java -jar ${SCRIPT_PATH}/target/parquet-benchmarks.jar p*Read* "$@"
 echo "Cleaning up generated data"
-java -cp ${SCRIPT_PATH}/target/parquet-benchmarks.jar parquet.benchmarks.DataGenerator cleanup
\ No newline at end of file
+java -cp ${SCRIPT_PATH}/target/parquet-benchmarks.jar org.apache.parquet.benchmarks.DataGenerator cleanup
